Services/CO.CDP.OrganisationInformation.Persistence/DatabaseTenantRepository.cs

```diff
@@ -27,6 +27,14 @@ public void Save(Tenant tenant)
         return await context.Tenants.FirstOrDefaultAsync(t => t.Name == name);
     }
 
+    public async Task<Person?> FindByUserUrn(string userUrn)
+    {
+        return await context.Persons
+                    .Include(p => p.Tenants)
+                    .Include(p => p.Organisations)
+                    .FirstOrDefaultAsync(p => p.UserUrn == userUrn);
+    }
```

Comment:
To comply with the above constraint, we need to query for tenants joined with persons, and then joined with organisations and organisation_persons (**not** joining back to persons from organisation_persons to complete the link).

![image](https://github.com/cabinetoffice/GCGS-Central-Digital-Platform/assets/190447/bc8544f6-b54a-4555-8964-eaba4b60438c)


If I was to write an SQL query, it would probably look like the following:

```sql
SELECT * from "Tenants" AS t
INNER JOIN "TenantPerson" tp ON t."Id" = tp."TenantId"
INNER JOIN "Persons" AS p ON tp."PersonId" = p."Id"
INNER JOIN "Organisations" AS o ON t."Id" = o."TenantId"
INNER JOIN "OrganisationPerson" AS op ON t."Id" = op."OrganisationId"
WHERE p."UserUrn" = 'urn:fdc:gov.uk:2022:123123123';

-[ RECORD 1 ]----------+----------------------------------------------------------------
Id                     | 1
Guid                   | 1b938e1f-ce79-472b-b134-497c599ea342
Name                   | Acme LTD
CreatedOn              | 2024-06-10 14:00:34.685333+00
UpdatedOn              | 2024-06-10 14:00:34.685344+00
TenantId               | 1
PersonId               | 1
Id                     | 1
Guid                   | 3e7816a4-0fa7-4e2c-af3d-2143f7507933
LastName               | Zalas
Email                  | jakub.zalas@example.com
FirstName              | Jakub
UserUrn                | urn:fdc:gov.uk:2022:123123123
Phone                  | +44792512333211
CreatedOn              | 2024-06-10 14:00:15.047253+00
UpdatedOn              | 2024-06-10 14:00:15.047269+00
Id                     | 1
Guid                   | 240fc843-211d-4af6-8ad4-5eb67e96ac36
Name                   | Acme LTD
Roles                  | {3}
ContactPoint_Email     | acme@example.com
ContactPoint_Name      |
ContactPoint_Telephone |
ContactPoint_Url       |
TenantId               | 1
CreatedOn              | 2024-06-10 14:00:34.685226+00
UpdatedOn              | 2024-06-10 14:00:34.685241+00
PersonId               | 1
OrganisationId         | 1
```

Reply:
or maybe:

```sql
SELECT * from "Tenants" AS t
INNER JOIN "TenantPerson" tp ON t."Id" = tp."TenantId"
INNER JOIN "Persons" AS p ON tp."PersonId" = p."Id"
INNER JOIN "Organisations" AS o ON t."Id" = o."TenantId"
INNER JOIN "OrganisationPerson" AS op ON o."Id" = op."OrganisationId"
INNER JOIN "Persons" AS opp ON op."PersonId" = opp."Id"
WHERE p."UserUrn" = 'urn:fdc:gov.uk:2022:123123123';
```